crates/vm/db.rs

```diff
@@ -1,8 +1,11 @@
 use ethrex_common::types::BlockHash;
 use ethrex_storage::Store;
 
+use crate::backends::exec_db::ExecutionDB;
+
 #[derive(Clone)]
-pub struct StoreWrapper {
-    pub store: Store,
-    pub block_hash: BlockHash,
+#[allow(clippy::large_enum_variant)]
+pub enum StoreWrapper {
+    Store(Store, BlockHash),
+    Execution(ExecutionDB, BlockHash),
```

Comment:
Clippy suggested to add a box due to the different sizes of the enum. Should I go for the suggestion instead?

```suggestion
pub enum StoreWrapper {
    Store(Store, BlockHash),
    Execution(Box<ExecutionDB>, BlockHash),
}
```